packages/devtools_app/lib/src/flutter/table.dart

```diff
@@ -0,0 +1,395 @@
+/// width of each row in the table.
+class TreeTable<T extends TreeNode<T>> extends StatefulWidget {
+  const TreeTable({
+    Key key,
+    @required this.columns,
+    this.treeColumn,
+    @required this.data,
+    @required this.keyFactory,
+  }) : super(key: key);
+
+  /// The columns to show in this table.
+  final List<ColumnData<T>> columns;
+
+  /// The column of the table to treat as expandable.
+  final TreeColumnData<T> treeColumn;
+
+  /// The tree structure of rows to show in this table.
+  final T data;
+
+  /// Factory that creates keys for each row in this table.
+  ///
+  /// It uses [PageStorageKey] to restore the table's state when re-opening
+  /// its page.
+  final String Function(T data) keyFactory;
+
+  @override
+  _TreeTableState<T> createState() => _TreeTableState<T>();
+}
+
+class _TreeTableState<T extends TreeNode<T>> extends State<TreeTable<T>>
+    with TickerProviderStateMixin {
+  List<T> _flattenedList = [];
+
+  /// The width of each column in this table, including the
+  /// appropriate indentation for nested rows.
+  ///
+  /// This is computed in [_computeTableWidth].
+  List<double> columnWidths = [];
+  double get tableWidth => columnWidths.reduce((x, y) => x + y);
+
+  AnimationController _resizeController;
+  Animation<double> _resizeAnimation;
+  Tween<double> _widthTween;
+
+  /// The width to assume for columns that don't specify a width.
+  static const defaultColumnWidth = 500.0;
+
+  /// The maximum height to allow for rows in the table.
+  static const defaultRowHeight = 42.0;
+
+  @override
+  void initState() {
+    super.initState();
+    widget.data.expandCascading();
+    _flattenedList = _flattenExpandedTree();
+    columnWidths = _computeTableWidth();
+    _widthTween = Tween<double>(begin: tableWidth, end: tableWidth);
+    _resizeController = AnimationController(
+      vsync: this,
+      duration: const Duration(milliseconds: 200),
+    );
+    _resizeAnimation = CurvedAnimation(
+      parent: _resizeController,
+      curve: Curves.easeInOutCubic,
+    );
+  }
+
+  @override
+  void didUpdateWidget(TreeTable<T> oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    _flattenedList = _flattenExpandedTree();
+    refreshTree();
+  }
+
+  @override
+  void dispose() {
+    _resizeController.dispose();
+    super.dispose();
+  }
+
+  List<T> _flattenExpandedTree({T node, bool Function(T node) filter}) {
+    node ??= widget.data;
+    filter ??= (_) => true;
+    List<T> flattenedChildren = [];
+    if (filter(node)) {
+      flattenedChildren = [
+        for (var child in node.children)
+          ..._flattenExpandedTree(
+            node: child,
+            filter: filter,
+          ),
+      ];
+    }
+    return [node, ...flattenedChildren];
+  }
+
+  // Determines the width of all columns, including the columns that
+  // use indentation to show the nesting of the table.
+  List<double> _computeTableWidth() {
```

Comment:
I would expect a method named `_computeTableWidth` to return a single double, so I'd suggest renaming to `_computeColumnWidths` since that is what the method is doing.